scipy/ndimage/tests/test_interpolation.py

```diff
@@ -538,518 +625,584 @@ def test_map_coordinates_endianness_with_output_parameter(self):
-        assert_array_almost_equal(out[::2], [0, 4, 1, 3])
+        assert_array_almost_equal(out[::2], xp.asarray([0, 4, 1, 3]))
 
     @pytest.mark.parametrize('order', range(0, 6))
-    def test_affine_transform24(self, order):
+    def test_affine_transform24(self, order, xp):
         # consistency between diagonal and non-diagonal case; see issue #1547
-        data = np.array([4, 1, 3, 2])
+        data = xp.asarray([4, 1, 3, 2])
         with suppress_warnings() as sup:
             sup.filter(UserWarning,
                        'The behavior of affine_transform with a 1-D array .* '
                        'has changed')
-            out1 = ndimage.affine_transform(data, [2], -1, order=order)
-        out2 = ndimage.affine_transform(data, [[2]], -1, order=order)
+            out1 = ndimage.affine_transform(data, xp.asarray([2]), -1, order=order)
+        out2 = ndimage.affine_transform(data, xp.asarray([[2]]), -1, order=order)
         assert_array_almost_equal(out1, out2)
 
     @pytest.mark.parametrize('order', range(0, 6))
-    def test_affine_transform25(self, order):
+    def test_affine_transform25(self, order, xp):
         # consistency between diagonal and non-diagonal case; see issue #1547
-        data = np.array([4, 1, 3, 2])
+        data = xp.asarray([4, 1, 3, 2])
         with suppress_warnings() as sup:
             sup.filter(UserWarning,
                        'The behavior of affine_transform with a 1-D array .* '
                        'has changed')
-            out1 = ndimage.affine_transform(data, [0.5], -1, order=order)
-        out2 = ndimage.affine_transform(data, [[0.5]], -1, order=order)
+            out1 = ndimage.affine_transform(data, xp.asarray([0.5]), -1, order=order)
+        out2 = ndimage.affine_transform(data, xp.asarray([[0.5]]), -1, order=order)
         assert_array_almost_equal(out1, out2)
 
     @pytest.mark.parametrize('order', range(0, 6))
-    def test_affine_transform26(self, order):
+    def test_affine_transform26(self, order, xp):
         # test homogeneous coordinates
-        data = np.array([[4, 1, 3, 2],
-                            [7, 6, 8, 5],
-                            [3, 5, 3, 6]])
+        data = xp.asarray([[4, 1, 3, 2],
+                           [7, 6, 8, 5],
+                           [3, 5, 3, 6]])
         if (order > 1):
             filtered = ndimage.spline_filter(data, order=order)
         else:
             filtered = data
-        tform_original = np.eye(2)
-        offset_original = -np.ones((2, 1))
-        tform_h1 = np.hstack((tform_original, offset_original))
-        tform_h2 = np.vstack((tform_h1, [[0, 0, 1]]))
+        tform_original = xp.eye(2)
+        offset_original = -xp.ones((2, 1))
+
+        concat = array_namespace(tform_original, offset_original).concat
+        tform_h1 = concat((tform_original, offset_original), axis=1)  # hstack
+        tform_h2 = concat( (tform_h1, xp.asarray([[0.0, 0, 1]])), axis=0)  # vstack
```

Comment:
can we use [`xp.stack`](https://data-apis.org/array-api/latest/API_specification/generated/array_api.stack.html) for either of these?

Reply:
Possibly. Is something wrong with what's here now though?

Reply:
this is okay 👍 let's fix the whitespace though
```suggestion
        tform_h1 = concat((tform_original, offset_original), axis=1)  # hstack
        tform_h2 = concat((tform_h1, xp.asarray([[0.0, 0, 1]])), axis=0)  # vstack
```